cmd/k8s-operator/tsrecorder.go

```diff
@@ -331,27 +328,31 @@ func (r *TSRecorderReconciler) getNodeID(ctx context.Context, tsrName string) (i
 	}
 	var profile profile
 	if err := json.Unmarshal(profileBytes, &profile); err != nil {
-		return "", false, fmt.Errorf("failed to extract node profile info from secret %s: %w", secret.Name, err)
+		return "", false, fmt.Errorf("failed to extract node profile info from state Secret %s: %w", secret.Name, err)
 	}
 
 	ok = profile.Config.NodeID != ""
 	return tailcfg.StableNodeID(profile.Config.NodeID), ok, nil
 }
 
-func (r *TSRecorderReconciler) getDeviceInfo(ctx context.Context, tsrName string) (hostname string, ips []string, ok bool, err error) {
+func (r *TSRecorderReconciler) getDeviceInfo(ctx context.Context, tsrName string) (d tsapi.TailnetDevice, ok bool, err error) {
 	nodeID, ok, err := r.getNodeID(ctx, tsrName)
 	if !ok || err != nil {
-		return "", nil, false, err
+		return tsapi.TailnetDevice{}, false, err
 	}
 
 	// TODO(tomhjp): The profile info doesn't include addresses, which is why we
 	// need the API. Should we instead update the profile to include addresses?
 	device, err := r.tsClient.Device(ctx, string(nodeID), nil)
 	if err != nil {
-		return "", nil, false, fmt.Errorf("failed to get device info from API: %w", err)
+		return tsapi.TailnetDevice{}, false, fmt.Errorf("failed to get device info from API: %w", err)
 	}
 
-	return device.Hostname, device.Addresses, true, nil
+	return tsapi.TailnetDevice{
+		Hostname:   device.Hostname,
+		TailnetIPs: device.Addresses,
+		URL:        fmt.Sprintf("https://%s.%s", device.Hostname, r.magicDNSSuffix),
```

Comment:
Instead of doing this, can we just read the full MagicDNS name from the current profile in the state secret, like we already do with the addresses? That way we won't have to do the thing with local client status read on start as well